docs/adr/adr_backend_etl_and_hosting_for_StatViz.md

```diff
@@ -0,0 +1,237 @@
+
+   Visualizations are typically embedded via URLs. This design means users won't have dynamic control over these filter values within embedded visualizations. Consequently, embedding URL filters as query variables suffices without burdening the system with additional dimensions that offer no direct user benefit.
+
+3. **[Data Transformation] Nightly Transformations**:
+
+   With our visualizations accessible via a public endpoint and the anticipation of embedding them on public channels, there's potential for a rapid increase in backend requests. This might peak especially during promotional campaigns by our users. To prevent any disruption to the Boxtribute app:
+
+   - We propose hosting the public GraphQL API endpoint on a distinct Google service, separate from the Boxtribute backend.
+   - Data will undergo nightly transformation and be cached independently. Consequently, requests to the public endpoint will bypass resources that the Boxtribute app directly accesses.
+
+4. **[Data Transformation] Caching in DB**:
+   We aim to cache this data in dedicated database tables to avoid the added complexity of filesystem or Redis caching. This is cost-effective in comparison to Redis caching. A lingering question, however, is whether to house this data in a distinct database or an isolated MySQL instance on Google Cloud SQL to guarantee no performance disruptions to the Boxtribute app.
+
+5. **[Data Transformation] Incremental updates of the cache**
+   Our caching approach seeks to facilitate incremental updates. These updates will be determined by new daily entries to the history table. Nevertheless, there exists a potential risk of the cache and the database becoming unsynchronized. Such discrepancies might arise from deleted history entries, direct edits to the db or actions that do not save changes to the history table. To counteract this, we intend to introduce tests to cross-check the transformed cache data against the Boxtribute database content.
+
+6. **[Data Structure] No additional db tables are needed in the Boxtribute DB**
+
+   The history table is sufficiently detailed, enabling us even to recreate past data snapshots. Since we are aiming for a nightly transformations, transforming the db (e.g. recreating past snapshots) do not have to be optimized for runtime.
+
+   While there's an option to introduce shadow tables for core tracked data like stock and people, transformations would still be essential to tailor this data for varied visualizations. Thus, there's no compelling reason to modify the operational database structure of Boxtribute.
+
+   Nevertheless, we should do an audit to ascertain that all transactions are tracked within the history table. For instance, operations like assigning or removing tags from stock or people currently bypass the history table.
+
+## Next steps
```

Comment:
After some of these are clarified, we should also update the C4 diagrams to have an architectural overview :)